tests/platform_tests/api/test_watchdog.py

```diff
@@ -47,57 +50,82 @@ def conf(self, request, duthost):
 
         if platform in test_config and 'default' in test_config[platform]:
             config.update(test_config[platform]['default'])
-
         if platform in test_config and hwsku in test_config[platform]:
             config.update(test_config[platform][hwsku])
 
-        assert 'valid_timeout' in config
+        pytest_assert('valid_timeout' in config, "valid_timeout is not defined in config")
         # make sure watchdog won't reboot the system when test sleeps for @TEST_WAIT_TIME_SECONDS
-        assert config['valid_timeout'] > TEST_WAIT_TIME_SECONDS * 2
-
+        pytest_assert(config['valid_timeout'] > TEST_WAIT_TIME_SECONDS * 2, "valid_timeout {} seconds is too short".format(config['valid_timeout']))
         return config
 
-
+    @pytest.mark.dependency()
     def test_arm_disarm_states(self, duthost, localhost, platform_api_conn, conf):
         ''' arm watchdog with a valid timeout value, verify it is in armed state,
         disarm watchdog and verify it is in disarmed state
         '''
         watchdog_timeout = conf['valid_timeout']
-        actual_timeout = watchdog.arm(platform_api_conn, watchdog_timeout)
+        try:
+            actual_timeout = int(watchdog.arm(platform_api_conn, watchdog_timeout))
+        except:
+            pytest.fail("actual watchdog timeout value is not an integer!")
 
-        assert actual_timeout != -1
-        assert actual_timeout >= watchdog_timeout
-        assert watchdog.is_armed(platform_api_conn)
+        if self.expect(actual_timeout is not None, "Watchdog.arm is not supported"):
+            if self.expect(actual_timeout != -1, "Failed to arm the watchdog"):
+                self.expect(actual_timeout >= watchdog_timeout, "Actual watchdog {} seconds apears wrong, should be less than {} seconds".format(actual_timeout, watchdog_timeout))
 
-        assert watchdog.disarm(platform_api_conn)
-        assert not watchdog.is_armed(platform_api_conn)
+        watchdog_status = watchdog.is_armed(platform_api_conn)
+        if self.expect(watchdog_status is not None, "Failed to retrieve watchdog status"):
+            self.expect(watchdog_status is True, "Watchdog is not armed.")
+
+        try:
+            remaining_time = int(watchdog.get_remaining_time(platform_api_conn))
+        except:
+            pytest.fail("remaining watchdog timeout value is not an integer!")
```

Comment:
You can simplify this to one pytest_assert() call. See [here](https://github.com/Azure/sonic-mgmt/blob/master/tests/platform_tests/api/test_chassis.py#L54) for example which checks if the value is of type 'str'.